Homework strings

shlembo598 opened the pull request.

shlembo598 committed:
Homework strings

shlembo598 committed:
Update homework_test.go

BalunCourses commented on homework/strings/homework_test.go:
```diff
@@ -11,27 +12,66 @@ import (
 	data []byte
 	refs *int
-	// need to implement
+	rwmu *sync.RWMutex
```
Это не нужно было делать

shlembo598 replied:
Убрал

BalunCourses commented on homework/strings/homework_test.go:
```diff
@@ -11,27 +12,66 @@ import (
+	if *b.refs > 1 {
+		*b.refs--
+	} else {
+		b.data = nil
```
Я бы не занулял

shlembo598 replied:
Исправил

BalunCourses commented on homework/strings/homework_test.go:
```diff
@@ -11,27 +12,66 @@ import (
+	b.rwmu.Lock()
+	defer b.rwmu.Unlock()
+
+	if index < 0 || index >= len(b.data) {
```
Первую часть условия можно делать без блокировки

shlembo598 replied:
Вообще убрал mutex

BalunCourses commented on homework/strings/homework_test.go:
```diff
@@ -11,27 +12,66 @@ import (
+		return false
+	}
+
+	if *b.refs > 1 {
```
Можно в условии переиспользовать NewCOWBuffer

shlembo598 replied:
Добавил NewCOWBuffer

shlembo598 committed:
Fixes

1. Убран sync.RWMutex
2. Убрано зануление в Close()
3. Update теперь использует NewCOWBuffer в своей реализации
